### PR TITLE
[MM-14548] Added e2e test for RHS search result comments

### PR DESCRIPTION
**Pull request by thekiiingbob**
#### Summary
Add e2e test for RHS search result comments

#### Ticket Link
[JIRA Ticket](https://mattermost.atlassian.net/browse/MM-14548)
Fixes https://github.com/mattermost/mattermost-server/issues/10437

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: cypress/integration/search/results_post_comment_spec.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+            // * Search results should have our original message
+            cy.get('#search-items-container').find(postMessageText).should('have.text', `${message}\n`);
+
+            // 5. Click on the reply button on the search result
+            // This button is visible only on hover. As a result,
+            // we have to force click here because we cannot
+            // trigger a CSS hover state.
+            // See - https://github.com/cypress-io/cypress/issues/1485#issuecomment-386085145
+            cy.get(`#SEARCH_commentIcon_${postId}`).click({force: true});
```
Comment (saturninoabril) [inline cypress/integration/search/results_post_comment_spec.js]:
```suggestion
            cy.clickPostCommentIcon(postId, 'SEARCH');
```

Please use the command we have for it.  I agree on the comment but I guess we could omit it.

Comment (thekiiingbob) [reply cypress/integration/search/results_post_comment_spec.js]: @saturninoabril Bah! Missed that command! All fixed up now.

**Comment by cometkim** [comment]
BTW, we have `#search-items-container`, `#reply_textbox` and `#rhsContainer` in a case :sweat_smile:  feels like we really have to clean inconsistent selectors once

File: cypress/integration/search/results_post_comment_spec.js (hunk 1 above)
Comment (saturninoabril) [reply cypress/integration/search/results_post_comment_spec.js]: No worries.  We're just starting out build all those commands and might need to reorganize an better document in-code in the future.

**Comment by saturninoabril** [comment]
@cometkim That's true.  Some of those are being used by Selenium so we need to keep it temporarily until we fully migrated to Cypress. :)

**Pull request merged**